vllm/model_executor/model_optimizer/fused_op_generator_utils.py

```diff
@@ -0,0 +1,152 @@
+import functools
+from typing import Callable, Dict, List, Optional
+
+import torch
+import torch.utils.cpp_extension
+
+from vllm.logger import init_logger
+
+from .utils import get_function_schema
+
+logger = init_logger(__name__)
+
+
+def compose2(f: Callable, g: Callable) -> Callable:
+    """
+    Compose two functions.
+    """
+    return lambda *a, **kw: g(f(*a, **kw))
+
+
+def compose(fs: List[Callable]) -> Callable:
+    """
+    Compose a list of functions.
+    """
+    return functools.reduce(compose2, fs)
+
+
+def is_optional_arg(n: torch.fx.Node, arg_num: int) -> bool:
+    """
+    Determine if the arg_num-th argument of n is optional or not based on
+    the associated function schema.
+    """
+    s = get_function_schema(n)
+    if not s or arg_num >= len(s.arguments):
+        return False
+    return isinstance(s.arguments[arg_num].real_type, torch._C.OptionalType)
+
+
+def build_extension(lib_name: str,
+                    sources: List[str],
+                    opt: str = '-O2',
+                    extra_cflags: Optional[List[str]] = None,
+                    extra_ldflags: Optional[List[str]] = None,
+                    verbose: bool = False):
+    """
+    Given a list of cpp and cuda source files, build and load a pytorch
+    extension module with the given name.  Loaded ops will appear in the
+    torch.ops.{lib_name} namespace.
+    """
+    if not extra_cflags:
+        extra_cflags = []
+
+    if not extra_ldflags:
+        extra_ldflags = []
+
+    torch.utils.cpp_extension.load(
+        name=lib_name,
+        sources=sources,
+        extra_cflags=[opt, f'-DLIBRARY_NAME={lib_name}', *extra_cflags],
+        extra_ldflags=extra_ldflags,
+        verbose=verbose,
+        is_python_module=False,
+    )
+
+
+def arg_schema_type(n: torch.fx.node.Argument,
+                    add_prefix: bool = False) -> str:
+    """
+    Get the schema or C++ type for a fused op argument.
+    """
+    if isinstance(n, float):
+        return "float"
+    elif isinstance(n, int):
+        return "int"
+    else:
+        if n.type is not None:
+            ty = n.type.__name__
```

Comment:
I think this should be 
```
    if isinstance(n, float):
        ty = "float"
    elif isinstance(n, int):
        ty = "int"
    elif n.type is not None:
```
so that the proper C++/Python type is returned.